Mathlib/SetTheory/Ordinal/Principal.lean

```diff
@@ -8,7 +8,8 @@ import Mathlib.SetTheory.Ordinal.FixedPoint
 /-!
 # Principal ordinals
 
-We define principal or indecomposable ordinals, and we prove the standard properties about them.
+We define principal or indecomposable ordinals, i.e. ordinals closed under a binary operation, and
```

Comment:
What binary operation? Presumably not an arbitrary one?

Reply:
One can talk about different flavors of principal ordinals, with the most common/useful being principal additive and principal multiplicative. That said, I am interesting in proving the characterization of exponentially principal and Veblen principal ordinals as well (since these just correspond to the epsilon and gamma ordinals, respectively, which I have a PR open for).

Reply:
I think this is still an extremely confusing first line. How about spelling it out: "If `r` is a binary operation on ordinals then we say that an ordinal is r-principal if, when considered as a set of ordinals, it is closed under this binary operation", or something like that? My instinct is that the module docstring of a mathlib file should be as readable as possible for non-experts (e.g. someone like me who knows what an ordinal is but has never heard of principal ordinals).